ArchiSteamFarm/Core/Utilities.cs

```diff
@@ -267,6 +268,78 @@ public static class Utilities {
 		return true;
 	}
 
+	internal static async Task<bool> EnsureUpdateDirectoriesPurged(params string[] directories) {
+		if ((directories == null) || (directories.Length == 0)) {
+			throw new ArgumentNullException(nameof(directories));
+		}
+
+		foreach (string directory in directories.Where(Directory.Exists)) {
+			ASF.ArchiLogger.LogGenericInfo(Strings.UpdateCleanup);
+
+			// Apart from simple removal, we're also going to do two additional things:
+			// 1. We're going to retry if the directory is still being used
+			// 2. We're going to ensure the changes were synchronized to disk, aka fsync()
+			using FileSystemWatcher watcher = new(Path.Combine(directory, ".."), Path.GetFileName(directory));
+
+			watcher.NotifyFilter = NotifyFilters.DirectoryName;
+
+			TaskCompletionSource<bool> tcs = new();
+
+			watcher.Deleted += onDeleted;
+
+			watcher.EnableRaisingEvents = true;
+
+			bool deleted = false;
+
+			for (byte i = 1; (i <= WebBrowser.MaxTries) && Directory.Exists(directory); i++) {
+				if (i > 1) {
+					await Task.Delay(5000).ConfigureAwait(false);
+				}
+
+				try {
+					Directory.Delete(directory, true);
+				} catch (IOException e) when ((i < WebBrowser.MaxTries) && ((uint) e.HResult == 0x80070020)) {
+					// It's entirely possible that old process is still running, we allow this to happen and add additional delay
+					ASF.ArchiLogger.LogGenericDebuggingException(e);
+
+					continue;
+				} catch (Exception e) {
```

Comment:
Since there is a compiler warning, it can be changed to `IOException or UnauthorizedAccessException` (assuming the second one can actually occur and is something that ASF is ok with)

Reply:
I disagree with this, which is why it's on notice level and not warning, also, it's qodana, not roslyn itself.

Our intention is to catch all other exception types, not some selected ones we're aware of.